core/controllers/editor.py

```diff
@@ -638,10 +639,17 @@ def post(self, entity_type, entity_id):
         allowed_formats = ', '.join(
             list(feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.keys()))
 
-        # Verify that the data is recognized as an image.
-        file_format = imghdr.what(None, h=raw)
-        if file_format not in feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS:
-            raise self.InvalidInputException('Image not recognized')
+        if filename is not None and filename.endswith('.svg'):
```

Comment:
Is it possible to detect svgs instead based on the file content? Sometimes files are named wrongly, so we never take the filename as the source of truth.

Reply:
Hm, we could add a new field in the POST call for the MIME type (determined while the file is uploaded in the UI). The would be the source of truth of truth for SVGs apart from the filename. Would this work?

Reply:
No, the mime type can be faked. In general never trust any content that comes from the frontend; the backend should do its own self-contained validation.